aptos-move/move-examples/token_objects/sources/aptos_token.move

```diff
@@ -530,19 +583,104 @@ module token_objects::aptos_token {
         update_typed_property(creator, token, &key, value);
     }
 
+    // Collection accessors
+
+    inline fun borrow_collection<T: key>(token: &Object<T>): &AptosCollection {
+        let collection_address = object::object_address(token);
+        assert!(
+            exists<AptosCollection>(collection_address),
+            error::not_found(ECOLLECTION_DOES_NOT_EXIST),
+        );
+        borrow_global<AptosCollection>(collection_address)
+    }
+
+    public fun is_mutable_collection_description<T: key>(
+        collection: Object<T>,
+    ): bool acquires AptosCollection {
+        borrow_collection(&collection).mutable_description
+    }
+
+    public fun is_mutable_collection_uri<T: key>(
+        collection: Object<T>,
+    ): bool acquires AptosCollection {
+        borrow_collection(&collection).mutable_uri
+    }
+
+    // Collection mutators
+
+    inline fun authorized_borrow<T: key>(collection: &Object<T>, creator: address): &AptosCollection {
+        let collection_address = object::object_address(collection);
+        assert!(
+            exists<AptosCollection>(collection_address),
+            error::not_found(ECOLLECTION_DOES_NOT_EXIST),
+        );
+        assert!(
+            collection::creator(*collection) == creator,
+            error::permission_denied(ENOT_CREATOR),
+        );
+        borrow_global<AptosCollection>(collection_address)
+    }
+
+    public fun set_collection_description<T: key>(
+        creator: &signer,
+        collection: Object<T>,
+        description: String,
+    ) acquires AptosCollection {
+        let aptos_collection = authorized_borrow(&collection, signer::address_of(creator));
+        assert!(
+            aptos_collection.mutable_description,
+            error::permission_denied(EFIELD_NOT_MUTABLE),
+        );
+        collection::set_description(option::borrow(&aptos_collection.mutator_ref), description);
+    }
+
+    public fun set_collection_uri<T: key>(
+        creator: &signer,
+        collection: Object<T>,
+        uri: String,
+    ) acquires AptosCollection {
+        let aptos_collection = authorized_borrow(&collection, signer::address_of(creator));
+        assert!(
+            aptos_collection.mutable_uri,
+            error::permission_denied(EFIELD_NOT_MUTABLE),
+        );
+        collection::set_uri(option::borrow(&aptos_collection.mutator_ref), uri);
+    }
+
+    // Collection entry functions
+
+    inline fun collection_object(creator: &signer, name: &String): Object<AptosCollection> {
+        let collection_addr = collection::create_collection_address(&signer::address_of(creator), name);
+        object::address_to_object<AptosCollection>(collection_addr)
+    }
+
+    entry fun set_collection_description_call(
```

Comment:
This is no longer required now that https://github.com/aptos-labs/aptos-core/pull/7090 is getting merged soon. set_collection_description can become public fun. Similar comment for other functions here